### PR TITLE
samples: update deployment-azure-vote samples

### PR DESCRIPTION
**Pull request by Redent0r**
Update deployment samples to no longer depend on azure-vote since this image is no longer available. Instead use azure linux based nginx image for frontend and echo-hello for backend. 

Additionally, if front is able to talk to the back, we see a log in the frontend app that looks like this: `Backend is available, starting nginx...` and stays in running state. If it's not available, we see logs like these: 
```
Backend is not available, retrying... attempt (1/5)
Backend is not available, retrying... attempt (2/5)
```
And eventually the container crashes after ~25s. 

###### Merge Checklist  
- [x] Followed patch format from upstream recommendation: https://github.com/kata-containers/community/blob/main/CONTRIBUTING.md#patch-format
  - [x] Included a single commit in a given PR - at least unless there are related commits and each makes sense as a change on its own.
- [x] Aware about the PR to be merged using "create a merge commit" rather than "squash and merge" (or similar)
- [x] genPolicy only: Ensured the tool still builds on Windows
- [x] The `upstream/missing` label (or `upstream/not-needed`) has been set on the PR. 

###### Test Methodology

test run: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=650813&view=result [pass]

### CODE REVIEW

**Review by danmihai1 — CHANGES_REQUESTED** [review]

File: src/agent/samples/policy/yaml/deployment/deployment-back.yaml [hunk 1]
```diff
@@ -2,34 +2,42 @@
...
-              value: azure-vote-back
-          resources:
-            requests:
-              cpu: 100m
-              memory: 128Mi
-            limits:
-              cpu: 250m
-              memory: 256Mi
+        - name: backend
+          image: "marinerconfpodsdev.azurecr.io/hashicorp/http-echo:latest"
```
Comment (Redent0r) [inline src/agent/samples/policy/yaml/deployment/deployment-back.yaml:25]: We are now taking a dependency of this image, which is a mirror of https://github.com/hashicorp/http-echo : "a small go web server that serves the contents it was started with as an HTML page". This is just to represent a "backend". I looked for and couldn't find alternative images in https://explore.ggcr.dev/?repo=mcr.microsoft.com

File: src/agent/samples/policy/yaml/deployment/deployment-front.yaml [hunk 2]
```diff
@@ -2,56 +2,46 @@
...
-              value: "yes"
-          resources:
-            requests:
-              cpu: 100m
-              memory: 128Mi
-            limits:
-              cpu: 250m
-              memory: 256Mi
+        - name: frontend
+          image: "mcr.microsoft.com/azurelinux/base/nginx:1"
```
Comment (Redent0r) [inline src/agent/samples/policy/yaml/deployment/deployment-front.yaml:25]: We are now depending on `azurelinux/base/nginx` image

File: src/agent/samples/policy/yaml/deployment/deployment-front.yaml [hunk 3]
```diff
@@ -2,56 +2,33 @@
...
     spec:
-      nodeSelector:
```
[the remark this answers is not in the document]
Comment (Redent0r) [reply src/agent/samples/policy/yaml/deployment/deployment-front.yaml:18]: Updated!

File: src/agent/samples/policy/yaml/deployment/deployment-front.yaml [hunk 4]
```diff
@@ -2,56 +2,46 @@
...
-              cpu: 100m
-              memory: 128Mi
-            limits:
-              cpu: 250m
-              memory: 256Mi
+        - name: frontend
+          image: "mcr.microsoft.com/azurelinux/base/nginx:1"
           ports:
-            - containerPort: 6379
-              name: redis
```
Comment (danmihai1) [inline src/agent/samples/policy/yaml/deployment/deployment-front.yaml:36]: Are we testing the port name using a some other YAML files?

Comment (Redent0r) [reply src/agent/samples/policy/yaml/deployment/deployment-front.yaml:36]: Good catch, I missed that. But yes https://github.com/microsoft/kata-containers/blob/msft-main/src/agent/samples/policy/yaml/kubernetes/conformance/csi-hostpath-plugin.yaml#L238

**Review by danmihai1 — APPROVED** [review]

**Review by ms-mahuber — APPROVED** [review]

**Pull request merged**